python/ray/dataframe/dataframe.py

```diff
@@ -614,18 +614,69 @@ def gt(self, other, axis='columns', level=None):
         raise NotImplementedError("Not Yet implemented.")
 
     def head(self, n=5):
-        raise NotImplementedError("Not Yet implemented.")
+        """Get the first n rows of the dataframe.
+
+        Args:
+            n (int): The number of rows to return.
+
+        Returns:
+            A new dataframe with the first n rows of the dataframe.
+        """
+        sizes = ray.get(self._map_partitions(lambda df: df.size)._df)
+        new_dfs = []
+        i = 0
+        while n > 0 and i < len(self._df):
+            if (n - sizes[i]) < 0:
+                new_dfs.append(_deploy_func.remote(lambda df: df.head(n),
```

Comment:
At some point we may want to consider defining the remote functions up front because deploying them every time could be slow (or maybe just deploying them once and then caching the result), but this is an optimization for later.